<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sss322" />

<meta name="date" content="2025-02-05" />

<title>Problem Set 1</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PLSCI 4170 Quantitative Genetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Problem Set 1</h1>
<h4 class="author">sss322</h4>
<h4 class="date">2025-02-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myporject1/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250127code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250127)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250127code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250127)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsss9010QG2025sss322tree2150f604beac5f5d9067ef4cb7935a8f4302e672targetblank2150f60a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/sss9010/QG2025sss322/tree/2150f604beac5f5d9067ef4cb7935a8f4302e672" target="_blank">2150f60</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsss9010QG2025sss322tree2150f604beac5f5d9067ef4cb7935a8f4302e672targetblank2150f60a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/sss9010/QG2025sss322/tree/2150f604beac5f5d9067ef4cb7935a8f4302e672" target="_blank">2150f60</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Problemset1.Rmd</code>) and HTML
(<code>docs/Problemset1.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sss9010/QG2025sss322/blob/2150f604beac5f5d9067ef4cb7935a8f4302e672/analysis/Problemset1.Rmd" target="_blank">2150f60</a>
</td>
<td>
sss9010
</td>
<td>
2025-02-17
</td>
<td>
ps1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sss9010/QG2025sss322/2150f604beac5f5d9067ef4cb7935a8f4302e672/docs/Problemset1.html" target="_blank">2150f60</a>
</td>
<td>
sss9010
</td>
<td>
2025-02-17
</td>
<td>
ps1
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sss9010/QG2025sss322/blob/5b366e9191a3582a9fcaff883e9b4f4c500aeb68/analysis/Problemset1.Rmd" target="_blank">5b366e9</a>
</td>
<td>
sss9010
</td>
<td>
2025-02-08
</td>
<td>
PS1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sss9010/QG2025sss322/5b366e9191a3582a9fcaff883e9b4f4c500aeb68/docs/Problemset1.html" target="_blank">5b366e9</a>
</td>
<td>
sss9010
</td>
<td>
2025-02-08
</td>
<td>
PS1
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div
id="q1-as-a-plant-breeder-i-want-to-fix-an-allele-that-provides-resistance-to-an-important-crop-disease."
class="section level4">
<h4><strong>Q1 As a plant breeder I want to fix an allele that provides
resistance to an important crop disease.</strong></h4>
<p>Unfortunately, I do not have the ability to directly genotype the
causative mutation, but I have identified a genetic marker that is
closely linked the causative mutation. Based on mapping experiments the
recombination frequency between the marker (alleles: A &amp; a) and the
QTL (alleles: Q – resistant &amp; q – susceptible) is r = .25I have
crossed a homozygous resistant line to a homozygous susceptible line to
get an offspringwith the genotype: !AQ$ %aq (I will self this individual
to generate offspring segregating for resistance.</p>
</div>
<div
id="a.-if-i-select-only-lines-with-the-genotype-aa-what-expected-proportion-of-the-selected-lines-that-will-be-homozygous-for-the-favorable-qtl-qq-2-pts"
class="section level4">
<h4><strong>A. If I select only lines with the genotype AA, what
expected proportion of the selected lines that will be homozygous for
the favorable QTL (QQ)? (2 pts)</strong></h4>
<p>In the whole genotype:</p>
<p>Probability of AA is 0.25 (from the Punnett square)</p>
<p>Probability of AQ = P(no crossover) * 0.5 = (1 - 0.25) * 0.5 =
0.375</p>
<p>I want the find the probability of QQ given AA, through using the
joint probability.</p>
<p>P= P(QQ and AA) / P(AA) = (P(AQ) * P(AQ)) / P(AA)</p>
<p>P= (0.375 * 0.375)) / 0.25 = 0.5625</p>
<p>The expected proportion of QQ given AA is equal to 0.5625</p>
<p><strong>The probability that i get the genotype where both marker
alleles are A and there has been no recombination between the marker and
the qtl is 0.5625</strong></p>
</div>
<div
id="q2-similar-to-q1-i-have-identified-markers-linked-to-a-qtl-of-interest-and-i-now-want-to-practice-marker-assisted-selection-to-fix-the-favorable-allele-in-a-population."
class="section level4">
<h4><strong>Q2 Similar to Q1 I have identified markers linked to a QTL
of interest and I now want to practice marker assisted selection to fix
the favorable allele in a population. </strong></h4>
<p>In this scenario I have 2 markers linked on either side of the QTL.
Marker 1 (alleles A, a) has a recombination frequency r =.4 with the QTL
and Marker 2 (alleles B, b) has a recombination frequency r =.3 with the
QTL. As before I’ve crossed a line that is homozygous for the beneficial
allele (Q) and a line that is homozygous for the unfavorable allele (q):
)AQB+,aqb. I selfed the heterozygous line above to generate many
offspring.The QQ genotype has a value of a = 5 for the trait of
interest, the unfavorable genotype qq has a value of -a = -5 for the
trait of interest. The trait is additive (no dominance)</p>
</div>
<div
id="a.-as-a-breeder-i-want-to-fix-the-resistance-allele.-if-i-select-only-lines-with-the-genotype-aabb-what-is-the-expected-proportion-of-the-selected-lines-that-will-be-homozygous-for-the-favorable-qtl-qq-2-pts"
class="section level4">
<h4><strong>A. As a breeder I want to fix the resistance allele. If I
select only lines with the genotype AABB, what is the expected
proportion of the selected lines that will be homozygous for the
favorable QTL (QQ)? (2 pts)</strong></h4>
<p>P(QQ given AABB) = P(QQ and AABB) / P(AABB)</p>
<p><strong>Joint probability:</strong></p>
<p>P(AABB) = P(no crossover or 2 crossovers in 1st haplotype) * P(no
crossover or 2 crossovers in 2nd haplotype)</p>
<p>P(AABB)= (0.5 * 0.6 * 0.7 + 0.5 * 0.4 * 0.3) * (0.5 * 0.6 * 0.7 + 0.5
* 0.4 * 0.3) = 0.27*0.27= 0.0729</p>
<p><strong>Probability of QQ and AABB:</strong></p>
<p>Probability of QQ and AABB = 0.5 * (0.6 * 0.7) * 0.5 * (0.6 * 0.7) =
0.0441</p>
<p><strong>P(QQ given AABB) </strong></p>
<p>Probability of (QQ given AABB) = 0.0441/0.0729 = 0.6049383</p>
<p><strong>The proportion of lines that have QQ given AABB genotype is
0.61</strong></p>
</div>
<div
id="b.-what-is-the-expected-value-for-offspring-with-the-genotype-aabb-3-pts"
class="section level4">
<h4><strong>B. What is the expected value for offspring with the
genotype AABB? (3 pts)</strong></h4>
<p>P(QQ given AABB) = 0.7778 ^2 =0.605.</p>
<p>P(qq given AABB)= (P(2 crossovers have happened and AB)/P(AB))^2 =
(0.5 * 0.4 * 0.3) / (0.5 * 0.4 * 0.3 + 0.5 * 0.6 * 0.7)^2 =0.222 ^ 2 =
0.049</p>
<p>P(Qq given AABB) = 2 * 0.7778 * 0.2222 = 0.345</p>
<pre class="r"><code>(0.605*5) + (0.345*0) + (0.049*-5) </code></pre>
<pre><code>[1] 2.78</code></pre>
<p>The expected value when the genotype is AABB is 2.7</p>
</div>
<div
id="c.-what-is-the-expected-value-for-lines-with-the-genotype-aa-3-pts"
class="section level4">
<h4><strong>C. What is the expected value for lines with the genotype
AA? (3 pts)</strong></h4>
<p>Probability of QQ given AA = P(Q given A) * P(Q given A) = (P(Q and
A)/P(A))^2 = ((0.5 * (1 - 0.4)) / 0.5)^2 = (0.5 * 0.6 / 0.5)^2 =
0.36</p>
<p>Probability of qq given AA = P(q given A) * P(q given A) = (P(Q and
A)/P(A))^2 = ((0.5 * 0.4 / 0.5)^2 = (0.4)^2 = 0.15</p>
<p>Probability of Qq given AA is 2 * 0.4 * 0.6 = 0.48</p>
<pre class="r"><code>(0.36*5) + (0.48*0) + (0.16*-5) </code></pre>
<pre><code>[1] 1</code></pre>
<p>Expected value whenthe genotype is AA =1</p>
</div>
<div
id="d.-what-is-the-expected-value-for-lines-with-the-genotype-bb-3-pts"
class="section level4">
<h4><strong>D. What is the expected value for lines with the genotype
BB? (3 pts)</strong></h4>
<p>Probability of QQ given BB = P(Q given B) * P(Q given B) = (P(Q and
B)/P(B))^2 = ((0.5 * (1 - 0.3)) / 0.5)^2 = (0.5 * 0.7 / 0.5)^2 =
0.49</p>
<p>Probability of qq given BB = P(q given B) * P(q given B) = (P(q and
B)/P(B))^2 = ((0.5 * 0.3 / 0.5)^2 = (0.5 * 0.3 / 0.5)^2 = 0.09</p>
<p>Probability of Qq given BB is 2 * 0.7 * 0.3 = 0.42</p>
<pre class="r"><code>(0.49*5) + (0.42*0) + (0.09*-5) </code></pre>
<pre><code>[1] 2</code></pre>
<p>Expected value when the genotype is BB =2</p>
</div>
<div
id="e.-how-close-would-a-single-marker-need-to-be-in-order-for-marker-assisted-selection-to-as-effective-and-the-two-marker-example-above-2-pts"
class="section level4">
<h4><strong>E. How close would a single marker need to be in order for
marker assisted selection to as effective and the two marker example
above? (2 pts)</strong></h4>
<p>x= the 1-r value for the MAS to be as effective as two marker
selection</p>
<p>q= 1-x</p>
<p>5x^2 + 2x(1-x) * 0 + (1-x)^2 * (-5) = 2.78</p>
<p>5x^2 + 0 + -5+10x-5p^2 = 2.7</p>
<p>10x = 2.7+5 = 7.78</p>
<p>x= 0.78</p>
<p>r= 1-x= 0.22</p>
<p><strong>The recombination frequency of a marker has to be smaller
than 0.22 to be as effective as two marker selection in MAS.
</strong></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=Estonian_Estonia.utf8  LC_CTYPE=Estonian_Estonia.utf8   
[3] LC_MONETARY=Estonian_Estonia.utf8 LC_NUMERIC=C                     
[5] LC_TIME=Estonian_Estonia.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       cli_3.6.3         knitr_1.49        rlang_1.1.5      
 [5] xfun_0.50         stringi_1.8.4     promises_1.3.2    jsonlite_1.8.9   
 [9] workflowr_1.7.1   glue_1.8.0        rprojroot_2.0.4   git2r_0.35.0     
[13] htmltools_0.5.8.1 httpuv_1.6.15     sass_0.4.9        rmarkdown_2.29   
[17] evaluate_1.0.3    jquerylib_0.1.4   tibble_3.2.1      fastmap_1.2.0    
[21] yaml_2.3.10       lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1    
[25] compiler_4.4.2    fs_1.6.5          Rcpp_1.0.14       pkgconfig_2.0.3  
[29] rstudioapi_0.17.1 later_1.4.1       digest_0.6.37     R6_2.5.1         
[33] pillar_1.10.1     magrittr_2.0.3    bslib_0.8.0       tools_4.4.2      
[37] cachem_1.1.0     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
