---
title: "HW3"
author: 
  name: "sss322"
  affiliation: ""
date: "February 4, 2025"
output:
  html_document:
    fig_caption: yes
    fig_height: 6
    fig_width: 12
    highlight: haddock
    theme: cosmo
    toc: yes
    toc_float:
      collapsed: no
      toc_depth: 2
editor_options: 
  chunk_output_type: console
---
  
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(message=FALSE)
knitr::opts_chunk$set(cache=FALSE)
```

### Load packages first  
```{r load packages, message=TRUE,include=FALSE }
packages_used <- c("AlphaSimR", "tidyverse", "workflowr", "here")
ip <- installed.packages()
all_packages_installed <- TRUE
for (package in packages_used){
  if (!(package %in% ip[,"Package"])){
    print(paste("Please install package", package))
    all_packages_installed <- FALSE
  }
}#END packages_used
if (!all_packages_installed) stop("Need to install packages")

library(tidyverse)
```


### Set random seed  
```{r set random seed}
random_seed <- 45678
set.seed(random_seed)
```

# Homework  
Illustrate regression to the mean between parents and offspring using AlphaSimR  

1. You know how to create a population -- create a population of parents
```{r}
nFounders <- 100
nChr <- 10 # Number of chromosomes
nSitesPerChr <- 1000 # Number of segregating sites _per chromosome_
nQTLperChr <- 250 # Vary this parameter to get oligo- versus poly- genic traits


founderHaps <- AlphaSimR::runMacs(nFounders, nChr, nSitesPerChr) #Simulating haplotypes

SP <- AlphaSimR::SimParam$new(founderHaps)
SP$addTraitA(nQtlPerChr=nQTLperChr)
founders <- AlphaSimR::newPop(founderHaps, simParam=SP) #Simulating parents genotypes based on the haplotypes and the trait charactherisitcs 
```


2. You know how to get phenotypes from that population  
```{r}
founders <- AlphaSimR::setPheno(founders, varE=1, simParam = SP)
hist(founders@pheno)
```


3. You want to randomly mate that population to get progeny that will be (we assume)
regressed to the mean. 

```{r}
parents<- rep(1:100, 1)
crossPlan <- matrix(rep(sample(parents, 200, replace = TRUE), 1), ncol=2, byrow=T)
random_mate <- AlphaSimR::makeCross(founders, crossPlan, simParam = SP)
head(crossPlan)
```



4. Having made the progeny population, phenotype it also  

```{r}
random_mate  <- AlphaSimR::setPheno(random_mate, varE=1,simParam = SP)
hist(random_mate@pheno)
```


5. Use each row of the `crossPlan` to find the two parents and calculate their phenotypic mean  


```{r}

crossPlan <- as.data.frame(crossPlan)
colnames(crossPlan) <- c("P1", "P2")

crossPlan$p_pheno1 <- sapply(crossPlan$P1, function(pid) founders@pheno[[pid]])# pulling the parent1 pheno
crossPlan$p_pheno2 <- sapply(crossPlan$P2, function(pid) founders@pheno[[pid]]) # pulling the parent2 pheno

crossPlan$mp <- rowMeans(cbind(crossPlan$p_pheno1, crossPlan$p_pheno2))# finding the midparent pheno

pheno <- as.data.frame(cbind(crossPlan$mp, random_mate@pheno))
colnames(pheno)<- c("midparent", "f1")
```



6. Make a scatterplot of the progeny phenotypes against the parent mean phenotypes  

```{r}
ggplot(data=pheno, aes(x=midparent, y=f1)) +
  geom_point()
```



> What is the regression coefficient?  

```{r}
mod<- lm(pheno$midparent ~pheno$f1)
regcoef<- mod$coefficients[2]

cat(
  "The regression coefficient is ",
  regcoef
)
```


> How much closer to the mean are offspring, on average, compared to their parents?  

```{r}
pheno_mean<- mean(c(founders@pheno, random_mate@pheno))

pheno$midparent_dev <-  pheno$midparent - pheno_mean
pheno$f1_dev <- pheno$f1 - pheno_mean

mean_f1_dev<- abs(mean(pheno$f1_dev))
mean_midparent_dev<- abs(mean(pheno$midparent_dev))

dif<- mean_midparent_dev - mean_f1_dev # deifference between the distance offspring and the midparents mean devaiations from the overall mean
```

```{r}
cat(
  "The midparents deviation from the overall mean is",
  dif,
  "units greater than the progeny mean phenotype deviation."
)
```
