---
title: "Lab4HW"
author: 
  name: "sss322"
  affiliation: "Cornell"
date: "February 11, 2025"
output:
  html_document:
    fig_caption: yes
    fig_height: 6
    fig_width: 12
    highlight: haddock
    theme: cosmo
    toc: yes
    toc_float:
      collapsed: no
      toc_depth: 2
editor_options: 
  chunk_output_type: console
---
  
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(message=FALSE)
knitr::opts_chunk$set(cache=FALSE)
knitr::opts_chunk$set(tidy=TRUE)
knitr::opts_chunk$set(results="hold")
knitr::opts_chunk$set(collapse=TRUE)
knitr::opts_chunk$set(fig.show="hold")
knitr::opts_chunk$set(tidy.opts = list(wrap=FALSE))
```

## Objectives

### Script setup  
```{r Script setup, echo=FALSE, results="hide"}
ip <- installed.packages()
packages_used <- c("AlphaSimR", "tidyverse", "workflowr", "here")
ip <- installed.packages()
all_packages_installed <- TRUE
for (package in packages_used){
  if (!(package %in% ip[,"Package"])){
    print(paste("Please install package", package))
    all_packages_installed <- FALSE
  }
}#END packages_used
if (!all_packages_installed) stop("Need to install packages")

library(tidyverse)

here::i_am("analysis/Lab4_DataForBreedingValue.Rmd")

source(here::here("code","addToREADME.R"))
addToREADME(paste0("## ", rmarkdown::metadata$title), append=F)
addToREADME(c(date(), ""))

packages_info <- ip[packages_used, c("Package", "Version", "Built")]
addToREADME(c("The packages used in this script are:", 
              "Package, Version, Built"))
apply(packages_info, 1, function(vec) addToREADME(paste(vec, collapse=" ")))
addToREADME("")

random_seed <- 45678
set.seed(random_seed)
addToREADME(paste("The random seed is", random_seed))
```

### Script parameters  
```{r script parameters}
nFounders <- 100
nChr <- 7
segSites <- 140
nQTL <- 100
```
```{r script parameters to README, echo=FALSE, results="hide"}
addToREADME(c(
  paste("Number of founders ", nFounders), 
  paste("Number of chromosomes ", nChr), 
  paste("Number of QTL per chromosome", nQTL),
  paste("Number of segregating sites per chromosome", segSites), "")
)
```

## Genotypic value versus breeding value  

```{r Compare breeding value to genotypic value}
# Create haplotypes for founder population of outbred individuals
founderHaps <- AlphaSimR::runMacs(nInd=nFounders, nChr=nChr, segSites=segSites)
# Setup the genotype to phenotype mapping
SP <- AlphaSimR::SimParam$new(founderHaps)
meanDD <- 0.4
varDD <- 0.04

# Additive and dominance model
SP$addTraitAD(nQtlPerChr=nQTL, meanDD=meanDD, varDD=varDD)

# Create a new population of founders
founders <- AlphaSimR::newPop(founderHaps, simParam=SP)

# Compare genotypic and breeding values
corBVGV <- cor(AlphaSimR::gv(founders), AlphaSimR::bv(founders)) %>% round(3)
plot(AlphaSimR::gv(founders), AlphaSimR::bv(founders), pch=16, 
     xlab="Genotypic value", ylab="Breeding value", 
     main=paste0("BV vs GV: AD model, cor=", corBVGV), 
     cex.axis=1.3, cex.lab=1.3)

# Additive dominance and epistasis model
# "the relative value of additive-by-additive variance 
# compared to additive variance"
SP <- AlphaSimR::SimParam$new(founderHaps)
relAA <- 0.5
SP$addTraitADE(nQtlPerChr=nQTL, meanDD=meanDD, varDD=varDD, relAA=relAA)

# Create a new population of founders
founders <- AlphaSimR::newPop(founderHaps, simParam=SP)

# Compare genotypic and breeding values
corBVGV <- cor(AlphaSimR::gv(founders), AlphaSimR::bv(founders)) %>% round(3)
plot(AlphaSimR::gv(founders), AlphaSimR::bv(founders), pch=16, 
     xlab="Genotypic value", ylab="Breeding value", 
     main=paste0("BV vs GV: ADE model, cor=", corBVGV), 
     cex.axis=1.3, cex.lab=1.3)
```
```{r Compare breeding value to genotypic value to README, echo=FALSE, results="hide"}
addToREADME(c("Chunk Compare breeding value to genotypic value",
              paste("Dominance degree mean", meanDD),
              paste("Dominance degree variance", varDD),
              paste("Relative AxA variance", relAA), "")
            )
```

## Estimated versus analytical breeding value  

```{r Estimate breeding value}
# Error variance for phenotypic evaluations
varE <- 1
# Number of progeny for breeding value estimation
nProgeny1 <- 5
nProgeny2 <- 50

# Estimate breeding values  
# ind is the individual whose breeding value you want to estimate  
# pop is the population that individual is in  
# nProgeny is the number of progeny for the test  
# varE is the error variance with which phenotypes are evaluated
#      if the genotypic variance is 1 then varE=1 will give h2 = 0.5
estimateBV <- function(ind, pop, nProgeny, varE=1){
  # I'm going to cheat a little and assume we know the population mean exactly
  popMean <- AlphaSimR::gv(pop) %>% mean
  # Set up crossPlan to cross ind to random others nProgeny times
  crossPlan <- cbind(ind, sample(AlphaSimR::nInd(pop), nProgeny, replace=T))
  progeny <- AlphaSimR::makeCross(pop, crossPlan)
  progPheno <- AlphaSimR::setPheno(progeny, varE=varE, onlyPheno=T)
  return(2*(mean(progPheno) - popMean))
}

# estimate BV with a progeny test of nProgeny1
estimatedBV <- sapply(1:AlphaSimR::nInd(founders), estimateBV, 
                      pop=founders, nProgeny=nProgeny1, varE=varE)
# Compare estimated and analytical breeding values
plot(estimatedBV, AlphaSimR::bv(founders), pch=16, 
     xlab="Estimated value", ylab="Analytical value", 
     main=paste("Breeding value estimated from", nProgeny1, "Progeny"))

# estimate BV with a progeny test of nProgeny2
estimatedBV <- sapply(1:AlphaSimR::nInd(founders), estimateBV, 
                      pop=founders, nProgeny=nProgeny2, varE=varE)
# Compare estimated and analytical breeding values
plot(estimatedBV, AlphaSimR::bv(founders), pch=16, 
     xlab="Estimated value", ylab="Analytical value", 
     main=paste("Breeding value estimated from", nProgeny2, "Progeny"))
```
```{r Estimate breeding value to README, echo=FALSE, results="hide"}
addToREADME(c("Chunk Estimate breeding value",
              paste("Phenotypic evaluation with varE", varE),
              paste("Number of progeny for first BV estimate", nProgeny1),
              paste("Number of progeny for first BV estimate", nProgeny2), ""
              )
            )
```

## Round robin crossing design  
This is a simple design that ensures that all parents get used equally  
```{r Round robin}
makeRoundRobin <- function(pop, makeRandom=F){
  nInd <- AlphaSimR::nInd(pop)
  parOrder <- 1:nInd
  if (makeRandom) parOrder <- sample(parOrder)
  return(cbind(parOrder, parOrder[c(2:nInd, 1)]))
}

# Make a bunch of new lines
crossPlan <- makeRoundRobin(founders)
exptLines <- AlphaSimR::makeCross(founders, crossPlan)

cbind(exptLines@id, exptLines@mother, exptLines@father) %>% head
```

### Set up records
 
```{r Set up records}
nInd <- 100
varE <- 1
# Function to make a simple data structure out of a population
# AlphaSimR doesn't retain varE once you have setPheno, so supply it
makeRecFromPop <- function(pop, varE=1){
  dplyr::tibble(id=pop@id,
                seedPar=pop@mother,
                pollenPar=pop@father,
                pheno=AlphaSimR::pheno(pop),
                varE=varE)
}

# Run a phenotyping experiment and populate the `records` data structure
exptLines <- AlphaSimR::setPheno(exptLines, varE=varE, simParam=SP)
records <- makeRecFromPop(exptLines, varE=varE)
head(records)

# Compare genotypic and phenotypic values
corPhGV <- cor(AlphaSimR::gv(exptLines), AlphaSimR::pheno(exptLines)) %>%
  round(3)
plot(AlphaSimR::gv(exptLines), AlphaSimR::pheno(exptLines), pch=16, 
     xlab="Genetic value", ylab="Phenotype", 
     main=paste0("First Evaluation, cor=", corPhGV))

# Second evaluation
exptLines <- AlphaSimR::setPheno(exptLines, varE=varE, simParam=SP)
# Add the new records to the old ones
records <- dplyr::bind_rows(records, makeRecFromPop(exptLines, varE=varE))

# Compare genotypic and phenotypic values
corPhGV <- cor(AlphaSimR::gv(exptLines), AlphaSimR::pheno(exptLines)) %>%
  round(3)
plot(AlphaSimR::gv(exptLines), AlphaSimR::pheno(exptLines), pch=16, 
     xlab="Genetic value", ylab="Phenotype", 
     main=paste0("Second Evaluation, cor=", corPhGV))
```
```{r Set up records to README, echo=FALSE, results="hide"}
addToREADME(c("Chunk Set up records",
              paste("Number of lines", nInd),
              paste("Error variance", varE)
              )
            )
```

## Use records to estimate genotypic value from >1 observation
```{r Linear model estimation}
# BASIC LINEAR MODEL
lmEstGV <- lm(pheno ~ -1 + id, data=records)
gvEstimates <- coefficients(lmEstGV)

# Compare genotypic and **Combined** phenotypic values
corEstGV <- cor(AlphaSimR::gv(exptLines), gvEstimates) %>%
  round(3)
plot(AlphaSimR::gv(exptLines), gvEstimates, pch=16, 
     xlab="Genetic value", ylab="Phenotype", 
     main=paste0("Both Evaluations, cor=", corEstGV))
```

## Use records in two-stage selection

```{r Two stage selection}
nIndStage1 <- 400
nIndStage2 <- 100
varEstage1 <- 4
varEstage2 <- 1

progenyPerCross <- nIndStage1 / nrow(crossPlan)
# The production pipeline starts with a bunch of new lines
exptLines <- AlphaSimR::makeCross(founders, crossPlan, 
                                  nProgeny=progenyPerCross)

# Phenotypic evaluation of experimental lines
exptLines <- AlphaSimR::setPheno(exptLines, varE=varEstage1, simParam=SP)
records <- makeRecFromPop(exptLines, varE=varEstage1)

# Select among lines to advance to Stage 2
keep <- AlphaSimR::pheno(exptLines) %>% 
  order(decreasing=T) %>% .[1:nIndStage2]

# Phenotypic evaluation of Stage 2 lines
stage2Lines <- exptLines[keep]
stage2Lines <- AlphaSimR::setPheno(stage2Lines, varE=varEstage2, simParam=SP)
records <- dplyr::bind_rows(records, makeRecFromPop(stage2Lines,
                                                    varE=varEstage2))
records1<- records
str(records)
```
```{r Two stage selection to README, echo=FALSE, results="hide"}
addToREADME(c("Two-stage selection",
              paste("Number of Stage 1 lines", nIndStage1),
              paste("Number of Stage 2 lines", nIndStage2),
              paste("Error variance for Stage 1", varEstage1),
              paste("Error variance for Stage 2", varEstage2), ""
              )
            )
```

## Estimate gain from selection four ways

```{r Estimate gain from selection}
print("Gain from selection on BV where the reference pop is stage2Lines")
print(paste("Gain from selection", 
            round(mean(AlphaSimR::bv(stage2Lines)) -
                  mean(AlphaSimR::bv(exptLines)), 2)))

print("Gain from selection on BV where the reference pop is all exptLines")
print(paste("Gain from selection", 
            round(mean(AlphaSimR::bv(exptLines)[keep]) - 
                  mean(AlphaSimR::bv(exptLines)), 2)))

print("Gain from selection on **GV** where the reference pop is stage2Lines")
print(paste("Gain from selection", 
            round(mean(AlphaSimR::gv(stage2Lines)) - 
                  mean(AlphaSimR::gv(exptLines)), 2)))

print("Gain from selection on **GV** where the reference pop is all exptLines")
print(paste("Gain from selection", 
            round(mean(AlphaSimR::gv(exptLines)[keep]) - 
                  mean(AlphaSimR::gv(exptLines)), 2)))
```



















## **Homework**  
**1. Tweak the linear model so that it estimates, for example, the breeding values of the seed parent**    


```{r}
lmEstGV <- lm(pheno ~ -1 + seedPar, data=records)
bvEstimates <- coefficients(lmEstGV)
```


**2. Make plots comparing the breeding values thus obtained to those obtained by the `bv` function or by our own `estimateBV` function.** 

```{r}
corEstBV<- round(cor(bvEstimates, estimatedBV),3)
plot(bvEstimates, estimatedBV, pch=16, 
     ylab="bv function", xlab="our function", 
     main=paste0("Both Evaluations, cor=", corEstBV))
```


```{r}
bv_sp <- AlphaSimR::bv(founders)
corEstBV<- round(cor(bvEstimates, bv_sp),3)
plot(bvEstimates, bv_sp, pch=16, 
     ylab="bv function", xlab="our function", 
     main=paste0("Both Evaluations, cor=", corEstBV))
```


**3. tweak the linear model and use the records from the `Two stage selection` chunk to account for the different error variances from Stage 1 versus Stage 2 when you estimate the genotypic values for individuals.** 

```{r}
lmEstGV_w <- lm(pheno ~ -1 + id, weights = 1/varE, data=records)
gvwEstimates <- coefficients(lmEstGV_w )
```

```{r}
lmEstGV <- lm(pheno ~ -1 + id, data=records)
gvEstimates <- coefficients(lmEstGV)
```

```{r}
gv_exp<- AlphaSimR::gv(exptLines)
corEstGV_w<- round(cor(gv_exp, gvwEstimates),3)
corEstGV<- round(cor(gv_exp, gvEstimates),3)
```


**4. Correlate the genotypic values obtained with or without weighting to the simulated values obtained by the `AlphaSimR::gv` function.** 

```{r}
plot(gv_exp, gvEstimates, pch=16, 
     xlab="Genetic value", ylab="Phenotype", 
     main=paste0("Both Evaluations without weights, cor=", corEstGV))
```

```{r}
plot(gv_exp, gvwEstimates, pch=16, 
     xlab="Genetic value", ylab="Phenotype", 
     main=paste0("Both Evaluations with weights, cor=", corEstGV_w))
```



5. *BONUS* for everyone: There is random error associated with all of these
simulations. If you want to see how the breeding value from the linear model
compares with the `estimateBV` function, or how weighted versus unweighted
estimates of genotypic value correlate to the truth, you will have to do each
analysis multiple times and get means and standard deviations.


```{r}
nIndStage1 <- 400
nIndStage2 <- 100
varEstage1 <- 4
varEstage2 <- 1
corrs<- data.frame()


for (i in  1:100){
  progenyPerCross <- nIndStage1 / nrow(crossPlan)
# The production pipeline starts with a bunch of new lines
  exptLines <- AlphaSimR::makeCross(founders, crossPlan, 
                                  nProgeny=progenyPerCross)
  # Phenotypic evaluation of experimental lines
  exptLines <- AlphaSimR::setPheno(exptLines, varE=varEstage1, simParam=SP)
  records <- makeRecFromPop(exptLines, varE=varEstage1)
  # Select among lines to advance to Stage 2
  keep <- AlphaSimR::pheno(exptLines) %>% 
    order(decreasing=T) %>% .[1:nIndStage2]
  # Phenotypic evaluation of Stage 2 lines
  stage2Lines <- exptLines[keep]
  stage2Lines <- AlphaSimR::setPheno(stage2Lines, varE=varEstage2, simParam=SP)
  records <- dplyr::bind_rows(records, makeRecFromPop(stage2Lines,
                                                    varE=varEstage2))
  
  #estimating genotypic value withthe weights
  lmEstGV_w <- lm(pheno ~ -1 + id, weights = 1/varE, data=records)
  gvwEstimates <- coefficients(lmEstGV_w)
  #estimating genotypic value without weights
  lmEstGV <- lm(pheno ~ -1 + id, data=records)
  gvEstimates <- coefficients(lmEstGV)
  
  gv_exp<- AlphaSimR::gv(exptLines)
  corEstGV_w<- round(cor(gv_exp, gvwEstimates),3)
  corEstGV<- round(cor(gv_exp, gvEstimates),3)
  corrv<- cbind(corEstGV, corEstGV_w)



  corrs<- rbind(corrs, corrv)

}

colnames(corrs)<- c("GV corr", "GV corr weighted")

boxplot(corrs)
paste("Mean of GV correlation without weight",mean(corrs$`GV corr` ))

paste("Mean of GV correlation with weight", mean(corrs$`GV corr weighted`))


```



6. *BONUS* for stat heads.  You could run a linear model with both 
`seedPar` and `pollenPar` as factors. But that would lead to separate estimates 
of these effects when in fact the same individuals are both seed and pollen 
parents.  How could you implement a linear model that would estimate one 
effect per parent regardless of whether it was a seed or pollen parent?  


```{r}
record_p<- pivot_longer(cols = c("seedPar", "pollenPar"),  values_to = "parent", data = records1)
head(record_p)
```

```{r}
lmEstBV <- lm(pheno ~ -1 + parent , data=record_p)
bvEstimates <- coefficients(lmEstBV)
```

```{r}
corEstBV<- round(cor(AlphaSimR::bv(founders), bvEstimates),3)
plot(AlphaSimR::bv(founders), bvEstimates, pch=16, 
     xlab="bv function", ylab="our function", 
     main=paste0("Both Evaluations, cor=", corEstBV))
```




